src/hist/stack.py

```diff
@@ -74,7 +81,14 @@ def plot(self, **kwargs: Any) -> list[Hist1DArtists]:
             if all(getattr(h, "name", None) is not None for h in self):
                 kwargs["label"] = [h.name for h in self]  # type: ignore
 
-        return hist.plot.histplot(list(self), **kwargs)  # type: ignore
+        fig = plt.gcf()
+        ax = fig.add_subplot()
+        ret = hist.plot.histplot(list(self), ax=ax, **kwargs)
```

Comment:
This should be flipped. Someone might pass `ax` already, so we need to support that. What we could/should do is get the axis from `ret`, then set legend entries from each item's name. We also need to make sure the names are set (I think they currently are?).

Reply:
FYI, I have enabled users to pass axes into `s.plot`, similar to the `_plot_ratiolike` method.